server/notification-providers/teams.js

```diff
@@ -57,41 +60,114 @@ class Teams extends NotificationProvider {
+                                                        "width": "32px",
+                                                        "style": "Person",
+                                                        "url": "https://raw.githubusercontent.com/louislam/uptime-kuma/master/public/icon.png",
+                                                        "altText": "Uptime Kuma Logo"
+                                                    }
+                                                ]
+                                            },
+                                            {
+                                                "type": "Column",
+                                                "width": "stretch",
+                                                "items": [
+                                                    {
+                                                        "type": "TextBlock",
+                                                        "size": "Medium",
+                                                        "weight": "Bolder",
+                                                        "text": headerMessage,
+                                                    },
+                                                    {
+                                                        "type": "TextBlock",
+                                                        "size": "Small",
+                                                        "weight": "Default",
+                                                        "text": "Uptime Kuma Alert",
+                                                        "isSubtle": true,
+                                                        "spacing": "None"
+                                                    }
+                                                ]
+                                            }
+                                        ]
+                                    }
+                                ]
+                            },
+                            {
+                                "type": "FactSet",
+                                "separator": true,
+                                "facts": facts
+                            }
+                        ],
+                        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+                        "version": "1.5"
+                    }
+                }
+            ]
         };
+
+        if (dashboardUrl) {
+            payload.attachments.forEach(element => {
+                element.content.push({
+                    "type": "ActionSet",
+                    "actions": [
+                        {
+                            "type": "Action.OpenUrl",
+                            "title": "Visit Uptime Kuma",
+                            "url": dashboardUrl
+                        }
+                    ]
+                });
+            });
```

Comment:
Why `forEach`?
Am I missing something?

```suggestion
            payload.attachments[0].content.push({
                "type": "ActionSet",
                "actions": [
                    {
                        "type": "Action.OpenUrl",
                        "title": "Visit Uptime Kuma",
                        "url": dashboardUrl
                    }
                ]
            });
```

Reply:
I went for the more generic way, but you're right, the array index is even more readable.